### PR TITLE
issues/KUI-1176 syllabus bugfix

### PR DESCRIPTION
**Pull request by belanglos**
This PR fixes
* bug in creation of syllabusName (including refactoring and removal of duplicate code).
* that no activeSyllabus was selected if there is no selectedSemester.

### CODE REVIEW

File: public/js/app/components/SyllabusInformation.jsx [hunk 1]
```diff
@@ -1,39 +1,35 @@
...
 }
 
-export default SyllabusInformation
+const SyllabusInformationOrNull = ({ syllabusName }) => {
+  if (!syllabusName) return null
+
+  return <ActualSyllabusInformation syllabusName={syllabusName} />
+}
+
+export { SyllabusInformationOrNull as SyllabusInformation }
```
Comment (belanglos) [inline public/js/app/components/SyllabusInformation.jsx:35]: Curious what you think about this solution
I thought that calling the component `SyllabusInformationOrNull` for `SyllabusInformation` would be misleading. Rather the component `ActualSyllabusInformation` should be called `SyllabusInformation`, but then we would have a mismatch in file name <-> exported component. 

Note also the change to named export.

Comment (karlandindrakryggen) [reply public/js/app/components/SyllabusInformation.jsx:35]: First: I like the change to named export. What are the real advantages of using default export? Or why don't we always use named exports?

I'm not sure I think calling `SyllabusInformationOrNull` for `SyllabusInformation` would be misleading actually. For the "user"  `SyllabusInformation` is what the component actually is. The null check with an extra component is just an internal behaviour the "user" doesn't need to know about ("user"=the other component using this component). 

And with that said, the only like important thing is the name of the actual exported component.

**Review by karlandindrakryggen — APPROVED** [review]

File: public/js/app/components/SyllabusInformation.jsx (hunk 1 above)
Comment (axelbjo) [reply public/js/app/components/SyllabusInformation.jsx:35]: Just curious about the approach here.

Why can’t the functional component ActualSyllabusInformation just be called SyllabusInformation and either just return the html if syllabusName exists or  return (null) if syllabusName does not exist? Is it not possible to get rid of SyllabusInformationOrNull altogether?

Comment (belanglos) [reply public/js/app/components/SyllabusInformation.jsx:35]: Very good question!
This is certainly possible. This is a rather philosophical discussion about readability and the notion that "one function should do one thing", which - I think - is mentioned in `Clean Code` in the chapter about functions.

**Review by axelbjo — APPROVED** [review]

File: public/js/app/components/SyllabusInformation.jsx (hunk 1 above)
Comment (amirhossein-haerian) [reply public/js/app/components/SyllabusInformation.jsx:35]: I completely agree with Karl and think the approach that you have used is indeed the cleanest approach that could be used here.
right now there is nothing that mislead other developers. I also really liked the aliasing approach for exporting the component.
I used to handle this functionality (checking to show the component if is not null) inline in the other components and now I think this approach can help me to have cleaner code. 
I'll also check the mentioned part inside the clean code book which is valuable to read.

**Pull request merged**